src/passes/strings-encoding/StringEncoding.cpp

```diff
@@ -88,6 +91,21 @@ inline bool isSkip(const StringEncodingOpt& encInfo) {
   return std::get_if<StringEncOptSkip>(&encInfo) != nullptr;
 }
 
+GlobalVariable *extractGlobalVariable(ConstantExpr *Expr) {
+  while (Expr) {
+    if (Expr->getOpcode() == llvm::Instruction::IntToPtr ||
+        llvm::Instruction::isBinaryOp(Expr->getOpcode())) {
```

Comment:
This looks good for the moment, but let's add some context. This takes into account that there can be other operations in intermediate nesting levels, e.g.:
```
%swift.bridge* inttoptr (
  i64 or (
    i64 sub (
      i64 ptrtoint (
        [18 x i8]* @.str.4 to i64),
      i64 32),
    i64 -9223372036854775808)
  to %swift.bridge*)
```

It seems that (1) these are always binary expressions and that (2) use of a string like `@.str.4` here will always be nested in the first operand. I wonder if we should add an assertion that fails if these assumptions break.

Reply:
Correct. While this works for now, this indeed looks quite fragile, as pretty ad-hoc. I'd have some further testing downstream, before merging this (I originally thought of opening this as a draft). Also, agree on add the a reduced instance of the `%swift.bridge*` use as a test.

Reply:
This looks SIL-specific and used in the Swift runtime to represent a bridged type (as per [here](https://github.com/apple/swift/blob/98e65d015979c7b5a58a6ecf2d8598a6f7c85794/lib/IRGen/IRGenSIL.cpp#L6827-L6829)), so I think (1) and (2) hold so long as the current bridging between Swift and ObjC (thus setting the higher bits of `BridgeObjectPtrTy`) holds.